app/scripts/controllers/transactions/tx-state-manager.js

```diff
@@ -183,53 +184,54 @@ export default class TransactionStateManager extends EventEmitter {
     const snapshot = snapshotFromTxMeta(txMeta);
     txMeta.history.push(snapshot);
 
-    const transactions = this.getFullTxList();
+    const transactions = this.getTransactions({
+      filterToCurrentNetwork: false,
+    });
     const txCount = transactions.length;
     const { txHistoryLimit } = this;
 
-    // checks if the length of the tx history is
-    // longer then desired persistence limit
-    // and then if it is removes only confirmed
-    // or rejected tx's.
-    // not tx's that are pending or unapproved
+    // checks if the length of the tx history is longer then desired persistence
+    // persistence limit and then if it is removes the oldest confirmed or
+    // rejected tx. Pending or unapproved transactions will not be removed by
+    // this operation.
+    //
+    // TODO: we are already limiting what we send to the UI, and in the future
+    // we will send UI only collected groups of transactions *per page* so at
+    // some point in the future, this persistence limit can be lifted. When we
+    // do that I think we should figure out a better storage solution for
+    // transaction history entries.
```

Comment:
I'm not sure about this :thinking: Not that it matters that we're committed to every TODO comment, but having an arbitrary persistence limit seems like a reasonable measure even in the long term, to limit the use of disk space. Allowing users to peruse ancient transactions that are loaded from the network on-the-fly sounds pretty far outside of our core competencies. Services like Etherscan are better suited for that sort of thing.

Reply:
going to rephrase "lifted" to "adjusted"

Reply:
done in latest